lib/charms/tls_certificates_interface/v3/tls_certificates.py

```diff
@@ -803,7 +807,7 @@ def generate_ca(
         .public_key(private_key_object.public_key())  # type: ignore[arg-type]
         .serial_number(x509.random_serial_number())
         .not_valid_before(datetime.now(timezone.utc))
-        .not_valid_after(datetime.now(timezone.utc) + timedelta(days=validity))
+        .not_valid_after(datetime.now(timezone.utc) + timedelta(**{EXPIRATION_UNITS: validity}))
```

Comment:
Dear @saltiyazan 

Thank you very much for your truly fast response :-)

Just to confirm that I understood correctly: you suggest to change the `validity` number to tiny float.

Our problem is the following.
We would like have the pipeline running automated tests of our charm. We would like NOT to change any _code_ within these tests (as that's the [current workaround](https://github.com/canonical/opensearch-operator/pull/430) :-( )

We would like to have a way to be able to "accelerate" expliry in a way that the code is left intact -- thus the pipeline will be assured to be stable regardless of how you guys may change internal implementations.

Thus the suggestion of introducing an environmental variable, as a way to configure the running charm.
(Since this is primarily for testing purposes, no need to introduce it in actual charm config.)